common/src/database/sqlite.rs

```diff
@@ -165,7 +165,17 @@ fn row_to_subscription(row: &Row) -> Result<SubscriptionData> {
     let query: String = row.get("query")?;
 
     let content_format = ContentFormat::from_str(row.get::<&str, String>("content_format")?.as_ref())?;
-    let princs_filter = PrincsFilter::from(row.get("princs_filter_op")?, row.get("princs_filter_value")?)?;
+
+    let client_filter_op: Option<String> = row.get("client_filter_op")?;
+
+    let client_filter = match client_filter_op {
+        Some(op) =>  {
+            let client_filter_type: Option<_> = row.get("client_filter_type")?;
+            let client_filter_type = client_filter_type.unwrap_or("KerberosPrinc".to_owned());
```

Comment:
What would you say if I:
- updated the database column to use an enum type and set Kerberos as default; or
- implemented a default method for `ClientFilterType`, which can be used here when the database contains old empty values?

Reply:
> updated the database column to use an enum type and set Kerberos as default;

This is PostgreSQL-specific as there are no enums in SQLite. We could use a default value for a string field, but it does not feel right.

> implemented a default method for ClientFilterType, which can be used here when the database contains old empty values?

If `client_filter_op` is set, `client_filter_type` MUST be set EXCEPT if we are processing an "old" filter. In this case, `ClientFilter::from` could easily use an `unwrap_or_default()` on `ClientFilterType` as you say, but... there is actually a third option:
- Write the "legacy" value in the migration (i.e. put "KerberosPrinc" in `client_filter_type` if `client_filter_op` is not null in the migration).

Since this is only a "migration" issue, I think it should be handled properly in the migration code. We don't want to have to maintain weird cases caused by past migrations in the application code forever.